sylt/src/formatter.rs

```diff
@@ -54,16 +54,34 @@ fn write_parameters<W: Write>(
     Ok(())
 }
 
-fn write_blob_fields<W: Write>(
+fn write_blob_fields<T: Clone, W: Write>(
     dest: &mut W,
     indent: u32,
-    fields: Vec<(String, Expression)>,
+    fields: Vec<(String, T)>,
+    sub_write: fn(&mut W, u32, T) -> fmt::Result,
 ) -> fmt::Result {
-    for (field, expr) in fields {
-        write_indents(dest, indent)?;
-        write!(dest, "{}: ", field)?;
-        write_expression(dest, indent, expr)?;
-        write!(dest, ",\n")?;
+    write!(dest, " {{ ")?;
+    match fields.len() {
+        0 => {
+            write!(dest, " }}")?;
+        }
+        1 => {
+            let (field, expr) = fields[0].clone();
+            write!(dest, "{}: ", field)?;
+            sub_write(dest, indent, expr)?;
+            write!(dest, " }}")?;
+        }
+        _ => {
+            write!(dest, "\n")?;
+            for (field, t) in fields {
+                write_indents(dest, indent)?;
+                write!(dest, "{}: ", field)?;
+                sub_write(dest, indent, t)?;
+                write!(dest, ",\n")?;
+            }
+            write_indents(dest, indent - 1)?;
+            write!(dest, "}}")?;
+        }
```

Comment:
I think this writes trailing whitespace when there is two fields or more. Line 63 and 75.